### PR TITLE
colour picker has been added (experimental)

### PR DESCRIPTION
**Pull request by naveen17797**
colour picker has been made in this branch as experimental feature, since themes needs to be refactored a lot,this cant be merged until we update themes.
1.theme setter has been added
screenshot from 2018-02-22 16-39-18

### CODE REVIEW

**Comment by naveen17797** [comment]
screenshot from 2018-02-22 17-26-13
much better look

**Comment by aethelwulffe** [comment]
Hi Naveen,
This is pretty darn cool.  Should we add the "experimental" tag to this for now?

We need to figure out how to rebase your fork master branch, so that your dev branches don't have every commit in the history since it's creation.  Something isn't quite right here.

**Comment by naveen17797** [comment]
I have added experimental tag since i dont know about the ui in most of the pages,
Yes, the old commit messages are being displayed when i make a PR, i will look in to it.

File: library/headers.inc.php [hunk 1]
```diff
@@ -19,6 +19,21 @@
  */
 ?>
 <link rel="stylesheet" href="<?php echo $css_header;?>" type="text/css">
+<style type="text/css">
```
Comment (aethelwulffe) [inline library/headers.inc.php]: I am not sure where/when a stylesheet reference crept into this file, but the intention of this file is just a library of asset function calls.  Everything in this file should be a function that calls includes.  Nothing else.
$css_header variable should contain the file location for holding this stuff (I think).

Comment (naveen17797) [reply library/headers.inc.php]: We need to make .css files in to .php files to use php variables inside them, so that we can use the colors we choose

Comment (naveen17797) [reply library/headers.inc.php]: By adding this feature i didnt want to break anything inside the ui, so i separated the bg classes and echoed out with our php variables

Comment (aethelwulffe) [reply library/headers.inc.php]: I get why you did it.  I am just hoping you can find a better place to put it than this particular file...unless it turns out that it is really really really the best place, in which case it should just have a function to call the code file, not the code.

Comment (naveen17797) [reply library/headers.inc.php]: Im planning to remove the <link rel= "stylesheet" and replace it with style.php file, i need to know are we ready to remove themes and adopt the theme setter.

Comment (aethelwulffe) [reply library/headers.inc.php]: Yes...provided we have things that look good, and we have padding control too, then we are set to move forward.  We just can't have the project go into an unusable state during a transition.

Comment (naveen17797) [reply library/headers.inc.php]: I will continue it to make a simple single theme with customizable options and make it ready for merge as soon as possible.

Comment (aethelwulffe) [reply library/headers.inc.php]: Sounds great.  Sending you a collaborator invite.

**Comment by naveen17797** [comment]
@pri2si17-1997 the color picker varies for different browsers, so the alternative may be creating color picker on js.

**Comment by pri2si17-1997** [comment]
http://www.eyecon.ro/colorpicker/

See if it can be used.

**Comment by aethelwulffe** [comment]
I highly recommend using the browser color picker.  It isn't just a matter of application responsiveness (though everything helps) but also a package maintenance thing.  People can get over the difference in a color picker image in the documentation...which they almost certainly would not read anyway.

Is there an actual application code-side issue with using native color pickers via HTML5?
*Yes, I am an HTML5 advocate.

**Comment by naveen17797** [comment]
@pri2si17-1997 as @aethelwulffe  said, i think we could use default browser color picker for picking colors instead of using js for the color picking.

**Comment by pri2si17-1997** [comment]
Ok. I was just suggesting. :slightly_smiling_face:  We can use browser color picker.

**Comment by naveen17797** [comment]
i had the same idea as you said before, but i feel there wont be much difference except the fact that we could control that color picker, in case of browser color picker we could not

**Comment by naveen17797** [comment]
see #1044

**Pull request closed without merging**